File: examples/r/paper_trade.R
## Paper-trade differentiator demo. Run: Rscript paper_trade.R
library(wickraexchange)

ex <- wkex_paper(c(USDT = 100000), maker_bps = 1, taker_bps = 5, slippage_bps = 10)
cat("venue:", wkex_name(ex), "\n")
wkex_set_price(ex, "BTC/USDT", 20000)

order <- wkex_place_market(ex, "BTC/USDT", "buy", 1)
cat("filled at", order$average_price, "(status", order$status, ")\n")

cat("  BTC free:", wkex_balance(ex, "BTC"), "\n")
cat("  USDT free:", wkex_balance(ex, "USDT"), "\n")

File: examples/python/paper_trade.py
"""Paper-trade differentiator demo. Run: python paper_trade.py"""

import wickra_exchange as wx


def main() -> None:
    ex = wx.Exchange.paper({"USDT": 100_000.0}, maker_bps=1.0, taker_bps=5.0, slippage_bps=10.0)
    print("venue:", ex.name())
    ex.set_price("BTC/USDT", 20_000.0)

    order = ex.place_order(wx.OrderRequest.market_buy("BTC/USDT", 1.0))
    print(f"filled at {order['average_price']} (status {order['status']})")

    for asset, free in ex.balances().items():
        print(f"  {asset} free: {free}")

    for event in ex.poll_events():
        print("event:", event["type"])


if __name__ == "__main__":
    main()
